docs/book/src/cronjob-tutorial/testdata/finalizer_example.go

```diff
@@ -76,7 +75,7 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// The object is being deleted
 		if controllerutil.ContainsFinalizer(cronJob, myFinalizerName) {
 			// our finalizer is present, so lets handle any external dependency
-			if err := r.deleteExternalResources(cronJob); err != nil {
+			if err := r.delete(ctx, cronJob); err != nil {
```

Comment:
The idea is to show that by using the finalizer you can do operations before effectively deleting the resource.
So, I agree the example is not clear. What about  

```suggestion
                         // The following method ilustrate that by using the Finalizer 
                         // before we allow the removal of the CronJob 
                         // we can perform all required operations.
                         // e.g do a HTTP request or update any other required 
                         // resource  
			if err := r. deleteExternalResources(); err != nil {
			       return ctrl.Result{}, err
			}
			// After we do all required operations
			// and ensure that the required criteria
			// to delete is succffuly achieved then,
			// we can remove the finalizer and delete it. 
			// Note that you do not need to use the finalizer
			// to remove any resource which is owned by the
			// kind. Remember that we use by controllerutil.<method to set the ownership>
			// More info: <add here the link for the k8s docs that explain the cascate delation>
			....
```
Also, we can improve the following method to better demostrate it::  

```go
func (r *Reconciler) deleteExternalResources() error {
	//
	// delete any external resources associated with the cronJob
	//
	// Ensure that delete implementation is idempotent and safe to invoke
	// multiple times for same object.
}
```

Reply:
ok iagree